smart-contract/program/src/state.rs

```diff
@@ -476,15 +511,18 @@ impl CentralState {
         token_mint: Pubkey,
         authority: Pubkey,
         total_staked: u64,
-    ) -> Self {
-        Self {
+    ) -> Result<Self, ProgramError> {
+        Ok(Self {
             tag: Tag::CentralState,
             signer_nonce,
             daily_inflation,
             token_mint,
             authority,
+            creation_time: Clock::get()?.unix_timestamp,
```

Comment:
Given that it will be used as the point of reference for all the program might be more convenient to round it to 00:000 ?

```rust
    pub fn new(
        signer_nonce: u8,
        daily_inflation: u64,
        token_mint: Pubkey,
        authority: Pubkey,
        total_staked: u64,
    ) -> Result<Self, ProgramError> {
        let now = Clock::get()?.unix_timestamp;
        Ok(Self {
            tag: Tag::CentralState,
            signer_nonce,
            daily_inflation,
            token_mint,
            authority,
            creation_time: now - (now - SECONDS_IN_DAY as i64),
            total_staked,
            total_staked_snapshot: 0,
            last_snapshot_offset: 0,
        })
    }
```

Reply:
Could be, but I don't see any benefit in this - time is dependent on the timezone anyways, so it wouldn't be 00:00 in most of the world